src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchIndexer.cs

```diff
@@ -57,11 +61,15 @@ public IReadOnlyCollection<SearchIndexEntry> Extract(ResourceElement resource)
         {
             EnsureArg.IsNotNull(resource, nameof(resource));
 
-            var poco = resource.ToPoco();
+            Resource poco = resource.ToPoco();
 
             var entries = new List<SearchIndexEntry>();
 
-            var context = new FhirEvaluationContext(poco);
+            var context = new FhirEvaluationContext(poco)
+            {
+                // Allows the indexer to parse the 'resolve' function i.e. CarePlan.subject.where(resolve() is Patient)
```

Comment:
Is it possible to write an e2e test that makes use of this `ElementResolver`

Reply:
Resolve will be used in https://github.com/microsoft/fhir-server/pull/873 when I update to the 4.0.1 search params files. If it makes more sense I can merge this PR into that instead